### PR TITLE
Use true scalars + bug fixes

### PR DESCRIPTION
**Pull request by mzient**
Fix usage of scalars - convert from {1} tensors to {}.
Fix a bug in executor - attempting to execute subsequent ops in the same stage after a failure.
Fix a bug in CocoReader - remove erroneuous assertion.

Classification readers are left out - changing the labels from 1-element tensors to scalars has merit, but I think we might have gone a bit too far there to change it now.

Signed-off-by: Michał Zientkiewicz <mzient@gmail.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes bugs:
    * in executor - attempting to execute subsequent ops in the same stage after a failure.
    * in CocoReader - remove erroneuous assertion.
- Refactoring to improve handling of scalar values
    * allow 0D data as scalar argument inputs
    * random generators produce true scalars
    * arithmetic operators accept and produce true scalars
    * audio decoder produces sampling rate as a true scalar

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     * Changed some tensors to 0D
     * Adjust tests
     * Changed the condition in OpSpec to accept 0D
 - Affected modules and functionalities:
     * OpSpec
     * ArithmOps
     * AudioDecoder
     * Random number generating ops (all of them)
 - Key points relevant for the review:
     * N/A
 - Validation and testing:
     * Existing tests apply
 - Documentation (including examples):
     * Updated, where applicable

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by mzient** [comment]
!build

**Review by JanuszL — APPROVED** [review]

**Comment by mzient** [comment]
!build

**Comment by JanuszL** [comment]
Please fix LGTM allert.

File: dali/python/nvidia/dali/types.py [hunk 1]
```diff
@@ -349,6 +351,7 @@ def ConstantNode(device, value, dtype, shape, layout, **kwargs):
         def _type_from_value_or_list(v):
             if not isinstance(v, (list, tuple)):
                 v = [v]
+                shape = []
```
Comment (JanuszL) [inline dali/python/nvidia/dali/types.py:354]: Nor used?

**Review by JanuszL — APPROVED** [review]

**Comment by mzient** [comment]
!build

**Comment by mzient** [comment]
!build

File: dali/python/nvidia/dali/types.py (hunk 1 above)
Comment (mzient) [reply dali/python/nvidia/dali/types.py:354]: Used in line 403. This was actually a necessary fix.

Comment (JanuszL) [reply dali/python/nvidia/dali/types.py:354]: Correct me if I'm wrong, but this is the scope of `_type_from_value_or_list` function and you are not changing the value of the `shape` variable from the outer scope.

**Comment by mzient** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

**Comment by mzient** [comment]
!build

**Comment by mzient** [comment]
!build

**Comment by mzient** [comment]
!build

**Comment by mzient** [comment]
!build

**Comment by mzient** [comment]
!build

**Pull request merged**